### DISCUSSION TITLE
jaeger-ui search filter by duration is misleading

### DISCUSSION THREAD

**Discussion by JaredTan95** [jaegertracing/jaeger#4111] [Q&A]
### What happened?

I set `max duration` and `min duration` in jaeger-ui, but it seems not work well.
 
![image](https://user-images.githubusercontent.com/12468337/207249960-15a35e15-3dfd-496e-bb10-c9aada12821b.png)

### Steps to reproduce

`max duration` and `min duration` in jaeger-ui

### Expected behavior

works well

**Comment by clement2026** [comment]
It is likely that a span of `insight-server: gorm.Query` has a duration less than 10ms.

If any span of a trace satisfies the filter, the trace should appear in the search result. The code of memory storage and Elasticsearch storage can confirm this.

memory storage:
https://github.com/jaegertracing/jaeger/blob/3d67ac86f53866332568b0f542265b00d86407fc/plugin/storage/memory/memory.go#L246-L253

Elasticsearch storage:
https://github.com/jaegertracing/jaeger/blob/3d67ac86f53866332568b0f542265b00d86407fc/plugin/storage/es/spanstore/reader.go#L509-L564

**Comment by JaredTan95** [comment]
> It is likely that a span of insight-server: gorm.Query has a duration less than 10ms.

 makes me more confused.
From my understanding, the search result's duration range will be between `1ms` and `10ms`.

**Comment by clement2026** [comment]
As you see, there are 87 spans in the `insight-server: gorm.Query` trace. If one of these spans has a duration between 1ms and 10ms, the whole trace should appear in the search results. 
When jaeger processes your search request, it searches all spans, not all traces.

**Comment by yurishkuro** [comment]
Correct - duration filter applies to individual spans, not to the duration of the whole trace or the root span.

**Reply by JaredTan95** [reply]
BTW, `root span` 's duration represents the whole trace's duration, right? If `root span`'s duration match the range, so the traceId would not be `aggs` and would not be searched then.

**Reply by JaredTan95** [reply]
If we filter the `root span` by determining whether the `reference` value exists, can it fix the problem?  @ClementRepo :-P

e.g. add `exists` query in  https://github.com/jaegertracing/jaeger/blob/3d67ac86f53866332568b0f542265b00d86407fc/plugin/storage/es/spanstore/reader.go#L509-L564 

```json
  "query": {
    "bool": {
      "must_not": [
        {
          "exists": {
            "field": "references.traceID"
          }
        }
      ], 
...
```

**Reply by yurishkuro** [reply]
> BTW, root span 's duration represents the whole trace's duration, right?

Not always, you can have child spans finishing or even starting after the parent is finished. Trace duration in the results is computed as max-min timestamp across all spans.

**Reply by JaredTan95** [reply]
Yes, The asynchronous pattern is a typical case.
